sam_view.c

```diff
@@ -267,7 +340,8 @@ int main_samview(int argc, char *argv[])
         .subsam_frac = -1.,
         .library = NULL,
         .bed = NULL,
-        .multi_region = 0
+        .multi_region = 0,
+        .barcode_tag = strndup(DEFAULT_BARCODE_TAG, 2)
```

Comment:
bam_aux_get seems to expect a char of 2 bytes: `uint8_t *bam_aux_get(const bam1_t *b, const char tag[2])` . Should it be allocated or is the following OK?
```diff
typedef struct samview_settings {
     size_t remove_aux_len;
     char** remove_aux;
     int multi_region;
-    char* barcode_tag;
+    char barcode_tag[2];
 } samview_settings_t;
```

Reply:
A function doesn't actually take an array as parameter. It's more of an indication to the programmer that the function will use the first two chars.

Both options work (static and dynamic allocation), just be consistent with the way you manage that memory space. With a static array, you will have to use `memcpy`, instead of `strdup`.